src/jit/codegenxarch.cpp

```diff
@@ -1837,8 +1837,12 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
     {
         // All long enregistered nodes will have been decomposed into their
         // constituent lo and hi nodes.
+
+#ifdef LEGACY_BACKEND
         regPairNo targetPair = treeNode->gtRegPair;
         noway_assert(targetPair == REG_PAIR_NONE);
+#endif
```

Comment:
codegenxarch is RyuJit backend implementation.  Hence there is no place for #ifdef LEGACY_BACKEND code here.  Please delete lines 1841-1844